### PR TITLE
Merge logger with turns

### PR DESCRIPTION
**Pull request by gacikjakub**

### CODE REVIEW

**Comment by gacikjakub** [comment]
Build 45 Successful
Deployed under: merge_logger_with_turns.battleships.kuba005.nazwa.pl

File: backend/src/main/java/pl/krkteam/battleships/authentication/PlayerHolder.java [hunk 1]
```diff
@@ -8,16 +10,24 @@
 
     private Set<Player> players = new HashSet<>();
 
+    private static final Logger logger = LogManager.getLogger(PlayerHolder.class);
+
     public boolean addPlayer(Player player) {
         if (player == null) {
```
Comment (boczkas) [inline backend/src/main/java/pl/krkteam/battleships/authentication/PlayerHolder.java:16]: null checking is not nice

File: backend/src/main/java/pl/krkteam/battleships/authentication/PlayerHolder.java [hunk 2]
```diff
@@ -8,16 +10,24 @@
...
+        boolean result = players.add(player);
+        if (result) {
+            logger.info("Player " + player.getName() + " has been added to PlayerHolder");
+        } else {
+            logger.info("Player " + player.getName() + " is in PlayerHolder already");
+        }
+        return result;
     }
 
     public boolean isRegister(Player player) {
```
Comment (boczkas) [inline backend/src/main/java/pl/krkteam/battleships/authentication/PlayerHolder.java:28]: isRegistered

File: backend/src/main/java/pl/krkteam/battleships/authentication/RegistrationController.java [hunk 3]
```diff
@@ -1,30 +1,48 @@
...
 import java.util.Set;
 
 @CrossOrigin(origins = "http://localhost:8080")
 @RestController
 public class RegistrationController {
 
+    private static final Logger logger = LogManager.getLogger(RegistrationController.class);
+
+    @Autowired
+    Game game;
```
Comment (boczkas) [inline backend/src/main/java/pl/krkteam/battleships/authentication/RegistrationController.java:22]: inject it by constructor

File: backend/src/main/java/pl/krkteam/battleships/authentication/RegistrationController.java [hunk 4]
```diff
@@ -1,30 +1,48 @@
...
 
-        System.out.println(playerDTO);
+        logger.debug("PlayerDTO: " + playerDTO);
 
         Player player = PlayerConversionUtil.convertPlayerDTOtoPlayer(playerDTO);
-
+    
         boolean result = playerHolder.addPlayer(player);
+        
+        game.getGameBoardHolder().addPlayer(player, new GameBoard());
```
Comment (boczkas) [inline backend/src/main/java/pl/krkteam/battleships/authentication/RegistrationController.java:40]: trains

File: backend/src/main/java/pl/krkteam/battleships/authentication/RegistrationController.java [hunk 5]
```diff
@@ -1,30 +1,48 @@
...
+    
+        game.getShotResultQueueHolder().addPlayer(player);
+    
+        game.getTurnHolder().addPlayer(player);
+
 
         String send = "{\"result\":" + "\"" + result + "\"" + " }";
```
Comment (boczkas) [inline backend/src/main/java/pl/krkteam/battleships/authentication/RegistrationController.java:47]: resultString / result

File: backend/src/main/java/pl/krkteam/battleships/common/domain/Board.java [hunk 6]
```diff
@@ -19,12 +18,16 @@ public Cell putCoordinatesAndCell(Coordinates coordinates, Cell cell) {
         return coordinatesCellMap.put(coordinates, cell);
     }
 
+    public Cell getCoorValue(Coordinates coordinates) {
```
Comment (boczkas) [inline backend/src/main/java/pl/krkteam/battleships/common/domain/Board.java:21]: getCellForCoordinate

File: backend/src/main/java/pl/krkteam/battleships/common/domain/Board.java [hunk 7]
```diff
@@ -19,12 +18,16 @@ public Cell putCoordinatesAndCell(Coordinates coordinates, Cell cell) {
...
+        return coordinatesCellMap.get(coordinates);
+    }
+
     public BoardSize getBoardSize() {
         return boardSize;
```
Comment (boczkas) [inline backend/src/main/java/pl/krkteam/battleships/common/domain/Board.java:26]: make it immutable

File: backend/src/main/java/pl/krkteam/battleships/common/domain/Board.java [hunk 8]
```diff
@@ -19,12 +18,16 @@ public Cell putCoordinatesAndCell(Coordinates coordinates, Cell cell) {
...
     }
 
-    public Map<Coordinates, Cell> getCoordinatesCellMap() {
-        return Collections.unmodifiableMap(coordinatesCellMap);
+    Map<Coordinates, Cell> getCoordinatesCellMap() {
+        return coordinatesCellMap;
```
Comment (boczkas) [inline backend/src/main/java/pl/krkteam/battleships/common/domain/Board.java:30]: Should be still immutable (like it was).
If it needs to be modified, you should create separate method for it

File: backend/src/main/java/pl/krkteam/battleships/common/domain/Game.java [hunk 9]
```diff
@@ -1,11 +1,27 @@
...
+import org.springframework.stereotype.Component;
+import pl.krkteam.battleships.opponent.shot.response.domain.ShotResultQueueHolder;
+import pl.krkteam.battleships.turns.holding.TurnHolder;
 
+@Component
 public class Game {
 
     private GameBoardHolder gameBoardHolder = new GameBoardHolder();
 
     private ShotResultQueueHolder shotResultQueueHolder = new ShotResultQueueHolder();
```
Comment (boczkas) [inline backend/src/main/java/pl/krkteam/battleships/common/domain/Game.java:13]: consider change of getters to something different if it needs modifications

File: backend/src/main/java/pl/krkteam/battleships/common/domain/GameBoard.java [hunk 10]
```diff
@@ -22,14 +20,16 @@ public ShipHolder getShipHolder() {
     }
 
     public Ship createShip(List<Coordinates> coordinates) {
-        Ship ship = new Ship();
+        Ship ship = new Ship(shipHolder);
         coordinates.forEach(coor -> {
```
Comment (boczkas) [inline backend/src/main/java/pl/krkteam/battleships/common/domain/GameBoard.java:24]: extract method

File: backend/src/main/java/pl/krkteam/battleships/common/domain/GameBoard.java [hunk 11]
```diff
@@ -22,14 +20,16 @@ public ShipHolder getShipHolder() {
...
-
+            Mast mast = new Mast(ship);
+            board.putCoordinatesAndCell(coor, mast);
         });
-        shipHolder.addShip(ship);
         return ship;
     }
+
+    public void reset(){
+        board=new Board(new BoardSize(10));
```
Comment (boczkas) [inline backend/src/main/java/pl/krkteam/battleships/common/domain/GameBoard.java:32]: spaces

**Pull request closed without merging**